boards/rpi-picow/drivers/led.c

```diff
@@ -0,0 +1,36 @@
+/**
+ * Initial RPi Pico W Board LED implementation
+ *
+ * Currently prints function calls at the moment.
+ *
+ * Copyright (C) 2022 Krzysztof Cabaj <kcabaj@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup boards_rpi_pico_w
+ * @{
+ * @file
+ * @author  Krzysztof Cabaj <kcabaj@gmail.com>
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "board.h"
+
+void _rpi_pico_w_LED0_OFF(void)
+{
+    printf("LED0_OFF\n");
+}
+
+void _rpi_pico_w_LED0_ON(void)
+{
+    printf("LED0_ON\n");
+}
+
+void _rpi_pico_w_LED0_TOGGLE(void)
+{
+    printf("LED0_TOGGLE\n");
+}
```

Comment:
just drop this, if the board has no LED, that's just the way it is.

Reply:
The board has LED0 but to power it on board MCU should send special command via SPI to the CYW32439, to which LED0 is physically connected. When PIO will be fully implements I plan to do this.

Reply:
IMHO that would be even worse.
The `LEDx_` macros are expected to be simple GPIO operations that can be used anywhere for debugging.
If this would involve complex SPI operations, this no longer holds.

Is the LED really controllable via the host MCU? I would have expected it's used by the CYW43439 firmware to e.g. signal connection state.

Reply:
I did not check this personally yet, but in the article [Raspberyy Pico W Micropython](https://how2electronics.com/getting-started-with-raspberry-pi-pico-w-using-micropython/)  they show blink program, and in included driver for CYW43439, there is a dedicated function to GPIO [set and get functions](https://github.com/georgerobotics/cyw43-driver/blob/195dfcc10bb6f379e3dea45147590db2203d3c7b/src/cyw43_ll.c) (`cyw43_ll_gpio_set` and `cyw43_ll_gpio_get`). So I'm pretty sure that MCU can control this LED, of course via CYW43439 firmware.

I agree that, using this for debug purposes sometimes could cause more problems. But, from my first experience with MCUs, LED is the first device used by newbies. Are there any other similar boards that do not have `LEDx_` macros but dedicated driver? If there is no strong disagree I would like to left these mockups and implement real LED control functions when PIO will be available. Of course in such a case I will add appropriate comment in the doxygen and board documentation.

Reply:
Just drop the mocks.
There are cases where you might not even have stdio (e.g. the bootloader) but still want to blink LEDs when available.

We don't really have a way to express dependencies for LEDs anyway, so you would always have to include the CYW43439 driver then.

Reply:
Ok, I understand your point of view. I will drop this code during refactoring.